docs/mathematics/ackermann-function.mdx

```diff
@@ -0,0 +1,138 @@
+---
+title: Ackermann Function
+slug: /mathematics/Ackermann-Function
+---
+
+## Introduction
+
+Ackermann function is one of the simplest examples of a total computable function,
+that is not primitive recursive.
+
+## Explanation
+
+It is basically a function with two non-negative integers as arguments.
+
+Ackermann function can be declared as:
+```
+A(0, j)=j+1 for j ≥ 0
+A(i, 0)=A(i-1, 1) for i > 0
+A(i, j)=A(i-1, A(i, j-1)) for i, j > 0
+```
+
+Let's have few examples: 
+```
+For i=1,j=2
+Output=4
+
+For i=3,j=3
+Output=61
+```
+
+## Algorithm
+
+```
+Ackermann_Function(int i, int j):
+  - Checking whether `i` is `0`,if yes 
+    then applying the associated condition.
+  - Checking whether `i` is greater than `0` and 
+    `j` is `0` , if yes then calling the function
+    with `m-1` and `1`.
+  - Checking whether `i` and `j` are greater than `0`, if yes
+    then applying the associated condition.
+```
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Dart', value: 'dart', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+int Ackermann_Function(int m, int n) {
+	if (m == 0)
+		return n + 1;
+
+	else if (m > 0 && n == 0)
+		return Ackermann_Function(m - 1, 1);
+
+	else if (m > 0 && n > 0)
+		return Ackermann_Function(m - 1, Ackermann_Function(m, n - 1));
+}
+
+int main() {
+	int m, n;
+	cout << "Input first Number\n";
+	cin >> m;
+	cout << "Input second Number\n";
+	cin >> n;
+	cout << "Result: " << Ackermann_Function(m, n);
+	return 0;
+}
+```
+
+</TabItem>
+<TabItem value="dart">
+
+```dart
+import 'dart:io';
+
+int Ackermann_Function(int m, int n) {
+  if (m == 0)
+    return n + 1;
+  else if (m > 0 && n == 0)
+    return Ackermann_Function(m - 1, 1);
+  else if (m > 0 && n > 0) return Ackermann_Function(m - 1, Ackermann_Function(m, n - 1));
+
+  return 0;
+}
+
+void main() {
+  print("Input first Number");
+  int m = int.parse(stdin.readLineSync()!);
+  print("Input second Number");
+  int n = int.parse(stdin.readLineSync()!);
+  int result = Ackermann_Function(m, n);
+  print("Result: $result");
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+#### Input:
+
+```
+Input first Number
+1
+Input second Number
+2
+```
+
+#### Output:
+
+```
+Result: 4
+```
+
+## Complexity Analysis
+
+- Time  complexity - `O(mA(m, n))`
+- Space complexity - `O(m)`
+
+## Credits
+
+- [Ritvij Kumar Sharma](https://github.com/ritvij14) contributed the [Dart implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Dart/maths/Ackermann_function.dart)
+- [Rajiv Singh](https://github.com/iamrajiv) contributed the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/math/Ackermann_function.cpp)
```

Comment:
Do a `git blame` to see who wrote the code. Rajiv has no implemented it.

Reply:
Done.